connectors/amazon/translator-dynamodb/src/main/java/org/teiid/translator/dynamodb/DynamoDBConnectionFactory.java

```diff
@@ -0,0 +1,50 @@
+package org.teiid.translator.dynamodb;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClient;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
+import org.teiid.translator.TranslatorException;
+
+public class DynamoDBConnectionFactory {
+    private final DynamoDBConfiguration dynamoDBConfiguration;
+    private final AWSStaticCredentialsProvider awsStaticCredentialsProvider;
+    private AmazonDynamoDB dynamoDBClient;
+
+
+    public DynamoDBConnectionFactory(DynamoDBConfiguration dynamoDBConfiguration) throws TranslatorException {
+        this.dynamoDBConfiguration = dynamoDBConfiguration;
+
+        if(dynamoDBConfiguration.getAccessKey() == null) {
+            throw new TranslatorException("Access key can't be null.");
+        }
+        if(dynamoDBConfiguration.getSecretKey() == null) {
+            throw new TranslatorException("Secret key can't be null");
+        }
+
+        AWSCredentials credentials = new BasicAWSCredentials(dynamoDBConfiguration.getAccessKey(), dynamoDBConfiguration.getSecretKey());
+        awsStaticCredentialsProvider = new AWSStaticCredentialsProvider(credentials);
+
+
+    }
+
+    public AmazonDynamoDB getDynamoDBClient() {
+        if (dynamoDBClient == null) {
+            synchronized (this) {
+                if (dynamoDBClient == null) {
+                    dynamoDBClient = AmazonDynamoDBClient
```

Comment:
Here I have a question. We can go two ways either with AmazonDynamoDBClient or with AmazonDynamoDBAsyncClient.

The AmazonDynamoDBAsyncClient follows async programming as the name suggests.
Ref: https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/basics-async.html

Let me know, which way to move forward.

Reply:
Start with the sync version as that matches what was done in simpledb.

You can look at doing the async later if you want once you have a better handle on things.  We have a simple async programming model.  See the DataNotAvailableException and it's usage in the Salesforce, Cassandra, or loopback translators along with http://teiid.github.io/teiid-documents/master/content/dev/Executing_Commands.html

Essentially if the engine asks for data from the translator and it does not have it, then you can have the translator throw an exception that may indicate a polling value.  If data arrives sooner than the polling value, the translator can notify the engine via ExecutionContext.dataAvailable.